File: C++/프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <vector>

using namespace std;

int dfs(int depth, int answer, vector<int> numbers, int target)
{
	if (depth == numbers.size())
	{
		if (answer == target) return 1;
		else return 0;
	}

	int number = numbers[depth];

	return dfs(depth + 1, answer + number, numbers, target) + dfs(depth + 1, answer - number, numbers, target);
}

int solution(vector<int> numbers, int target)
{
	int answer = 0;
	int num = numbers[0];

	answer = dfs(1, num, numbers, target) + dfs(1, -num, numbers, target);

	return answer;
}

File: Python3/프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(cur, op, numbers, target):
    if cur == len(numbers):
        return 1 if op == target else 0
    
    nxt = cur + 1;
    return dfs(nxt, op + numbers[cur], numbers, target) + dfs(nxt, op - numbers[cur], numbers, target)

def solution(numbers, target):
    answer = dfs(1, numbers[0], numbers, target) + dfs(1, -numbers[0], numbers, target)
    
    return answer